test/common/cdp-driver.js

```diff
@@ -0,0 +1,238 @@
+            frameNameToFrameId[info.frame.name] = info.frame.id;
+
+        // were we waiting for this frame to be loaded?
+        if (frameWaitPromiseResolve && frameWaitName === info.frame.name) {
+            frameWaitPromiseResolve();
+            frameWaitPromiseResolve = null;
+        }
+    });
+
+    // track execution contexts so that we can map between context and frame IDs
+    client.Runtime.executionContextCreated(info => {
+        debug("executionContextCreated " + JSON.stringify(info));
+        frameIdToContextId[info.context.auxData.frameId] = info.context.id;
+    });
+
+    client.Runtime.executionContextDestroyed(info => {
+        debug("executionContextDestroyed " + info.executionContextId);
+        for (let frameId in frameIdToContextId) {
+            if (frameIdToContextId[frameId] == info.executionContextId) {
+                delete frameIdToContextId[frameId];
+                break;
+            }
+        }
+    });
+}
+
+
+// helper functions for testlib.py which are too unwieldy to be poked in from Python
+function getFrameExecId(frame) {
+    var frameId = frameNameToFrameId[frame];
+    if (!frameId)
+        throw Error(`Frame ${frame} is unknown`);
+    var execId = frameIdToContextId[frameId];
+    if (!execId)
+        throw Error(`Frame ${frame} (${frameId}) has no executionContextId`);
+    return execId;
+}
+
+function expectLoadFrame(name, timeout) {
+    return new Promise((resolve, reject) => {
+        let tm = setTimeout( () => reject("timed out waiting for frame load"), timeout );
+
+        // we can only have one Page.frameNavigated() handler, so let our handler above resolve this promise
+        frameWaitName = name;
+        new Promise((fwpResolve, fwpReject) => { frameWaitPromiseResolve = fwpResolve })
+            .then(() => {
+                // For the frame to be fully valid for queries, it also needs the corresponding
+                // executionContextCreated() signal. This might happen before or after frameNavigated(), so wait in case
+                // it happens afterwards.
+               function pollExecId() {
+                    if (frameIdToContextId[frameNameToFrameId[name]]) {
+                        clearTimeout(tm);
+                        resolve();
+                    } else {
+                        setTimeout(pollExecId, 100);
+                    }
+                }
+                pollExecId();
+            });
+    });
+}
+
+/**
+ * SSL handling
+ */
+
+// secure by default; tests can override to "continue"
+// https://chromedevtools.github.io/devtools-protocol/1-3/Security/#type-CertificateErrorAction
+var ssl_bad_certificate_action = "cancel";
+
+function setupSSLCertHandling(client) {
+    client.Security.enable();
+
+    client.Security.setOverrideCertificateErrors({override: true});
+    client.Security.certificateError(info => {
+        process.stderr.write(`CDP: Security.certificateError ${JSON.stringify(info)} (ignoring)\n`);
```

Comment:
That's a leftover from the previous state. It should say `(${ssl_bad_certificate_action})`, not `(ignoring)`, as this now does *not* ignore SSL errors by default.